tests/integration_test/core_mqtt_system_test.c

```diff
@@ -419,6 +459,64 @@ static void eventCallback( MQTTContext_t * pContext,
                            MQTTPacketInfo_t * pPacketInfo,
                            MQTTDeserializedInfo_t * pDeserializedInfo );
 
+/**
+ * @brief Helper function to subscribe to the given MQTT topic.
+ *
+ * @param[in] pContext MQTT context pointer.
+ * @param[in] pTopic MQTT topic to subscribe to.
+ * @param[in] qos MQTT subscribe packet quality of service.
+ *
+ * @return MQTTSuccess if successful, error code otherwise. See #MQTT_Subscribe
+ * for possible return codes.
+ */
+static MQTTStatus_t subscribeToTopic( MQTTContext_t * pContext,
+                                      const char * pTopic,
+                                      MQTTQoS_t qos );
+
+/**
+ * @brief Helper function to unsubscribe from the given MQTT topic.
+ *
+ * @param[in] pContext MQTT context pointer.
+ * @param[in] pTopic MQTT topic to unsubscribe from.
+ *
+ * @return MQTTSuccess if successful, error code otherwise. See #MQTT_Unsubscribe
+ * for possible return codes.
+ */
+static MQTTStatus_t unsubscribeFromTopic( MQTTContext_t * pContext,
+                                          const char * pTopic );
+
+/**
+ * @brief Helper function to publish to the given MQTT topic.
+ *
+ * @param[in] pContext MQTT context pointer.
+ * @param[in] pTopic MQTT topic to publish to.
+ * @param[in] pPayload Message payload.
+ * @param[in] setRetainFlag If this is true, then the retrain flag will be set.
+ * @param[in] isDuplicate If this is true, then the duplicate flag will be set.
+ * @param[in] qos The PUBLISH packet quality of service.
+ * @param[in] packetId A packet ID generated by #MQTT_GetPacketId.
+ *
+ * @return MQTTSuccess if successful, error code otherwise. See #MQTT_Publish
+ * for possible return codes.
+ */
+static MQTTStatus_t publishToTopic( MQTTContext_t * pContext,
+                                    const char * pTopic,
+                                    const char * pPayload,
+                                    bool setRetainFlag,
+                                    bool isDuplicate,
+                                    MQTTQoS_t qos,
```

Comment:
These are all of the fields for an `MQTTPublishInfo_t`, except for the sizes of the strings. At this point, would it just be easier to pass in the struct? We could even have a default initializer macro since most of these fields don't change.

Reply:
sure

Reply:
I don't think one is easier over the other lol